Add sections on hyperparameter tuning and cross-validation

maltelueken opened the pull request:
Closes #53 

Adds the following sections:

Hyperparameter tuning: 

- Manual
- Grid search
- Nested CV 
- Exercise

Selecting the best model:

- Train and test error
- Validation curves (relates to over and underfitting, bias-variance trade-off)

We still need to see how these should be ordered and adapted.

Changes the dataset in the exercise on learning and validation curves from blood transfusion to adult census. Participants are already familiar with the adult census dataset and this reduces the complexity of the exercise.

f-hafner commented on python_scripts/cross_validation_train_test.py:
```diff
@@ -0,0 +1,373 @@
+#
+# In the previous notebooks, we introduce some concepts regarding the evaluation
+# of predictive models. While this section could be slightly redundant, we
+# intend to go into details into the cross-validation framework.
```
```suggestion
# In the previous notebooks, we introduce some concepts regarding the evaluation
# of predictive models. Now, we discuss the details of the cross-validation framework.
```

maltelueken commented:
I reverted back to the blood transfusion dataset in the cross-validation exercise because it is much smaller than the adult census one which leads to much faster fitting times (fitting would take an hour with the entire adult census dataset).

I also fixed the deprecation warning from skicit-learn 1.6.
